metadata-ingestion/src/datahub/ingestion/source/aws/glue.py

```diff
@@ -795,6 +935,16 @@ def get_workunits(self) -> Iterable[MetadataWorkUnit]:
                 self.report.report_workunit(mcp_wu)
                 yield mcp_wu
 
+            mcps_profiling = self.get_profile_if_enabled(mce, database_name, table_name)
```

Comment:
why not check it here if profiling enabled and then you can call your method as `self.get_profile` and if it would return a list.
And use enumerate instead of `range(len(mcps_profiling))`
Something like this:
```
if self.config.profiling is not None:
    mcps_profliling = self.get_profile(mce, database_name, table_name)
    for mcp_index, mcp in enumerate(xs):
                mcp_wu = MetadataWorkUnit(
                        id=f"profile-{full_table_name}-partition-{mcp_index}",
                        mcp=mcp,
                    )
                self.report.report_workunit(mcp_wu)
                yield mcp_wu
```

Reply:
The indenting and variable names might not correct as I wrote here in github this code

Reply:
The reason for `get_profile_if_enabled` function is to keep in line with the existing `get_lineage_if_enabled` function. But I can definitely move the checking of self.config.profiling to ouside of the function. What do you think?

I will apply the enumerate change!